### DISCUSSION TITLE
Hook into title afterStateUpdated?

### DISCUSSION THREAD

**Discussion by howdu** [camya/filament-title-with-slug#3] [Q&A] [answered]
Is there a way of hooking into the afterStateUpdated for title I need to do some custom logic?

**Comment by camya** [comment]
Hi, can you give me a use case for this feature?

**Reply by howdu** [reply]
Thanks for your reply - I  wanted to set the value of another field when the title's state is changed.

We have a duplicate 'name' field which is for internal use only. As a quick fix I've done this with an observer instead.

**Comment by camya** [comment, marked as answer]
Hi, I added this feature and updated the [documentation](https://github.com/camya/filament-title-with-slug#all-available-parameters).

You can now use:

`titleAfterStateUpdated: function ($state) {},`

`slugAfterStateUpdated: function ($state) {},`

These closures are called at the end of afterStateUpdated() ... [see source](https://github.com/camya/filament-title-with-slug/commit/3bc78cb1159e6e83a79ff0fb7be7f63312f650d1#diff-9272c6723c95b6f07a39631f58a6743cdf0a736aef11526fba75324ddd527d90)
